js_modules/dagit/packages/core/src/ticks/SensorDryRun.tsx

```diff
@@ -0,0 +1,361 @@
+              </div>
+              <div>
+                <Subheading>Computed cursor value</Subheading>
+                <pre>
+                  {sensorExecutionData?.evaluationResult?.cursor?.length
+                    ? sensorExecutionData?.evaluationResult.cursor
+                    : error
+                    ? 'Error'
+                    : 'None'}
+                </pre>
+                {error ||
+                (currentCursor ?? '') ===
+                  (sensorExecutionData?.evaluationResult?.cursor ?? 'testing1234') ? null : (
+                  <Box flex={{direction: 'row', gap: 8, alignItems: 'center'}}>
+                    <Button
+                      disabled={['Persisting', 'Persisted'].includes(cursorState)}
+                      loading={cursorState === 'Persisting'}
+                      onClick={onPersistCursorValue}
+                    >
+                      <span data-testid={testId('persist-cursor')}>
+                        {cursorState === 'Persisting'
+                          ? 'Persisting'
+                          : cursorState === 'Persisted'
+                          ? 'Persisted'
+                          : 'Persist computed cursor value'}
+                      </span>
+                    </Button>
+                    {cursorState === 'Persisted' ? (
+                      <Icon name="check_circle" color={Colors.Green500} />
+                    ) : null}
+                  </Box>
+                )}
+              </div>
+            </Grid>
+            {error ? (
+              <div>
+                <PythonErrorInfo error={error} />
+              </div>
+            ) : null}
+            {didSkip ? (
+              <div>
+                <Subheading>Skip Reason</Subheading>
+                <div>{sensorExecutionData?.evaluationResult?.skipReason}</div>
+              </div>
+            ) : null}
+            {numRunRequests && runRequests ? (
+              <RunRequestTable
+                runRequests={runRequests}
+                name={name}
+                jobName={jobName}
+                isJob={true}
+                repoAddress={repoAddress}
+              />
+            ) : null}
+          </Box>
+        </Box>
+      );
+    }
+    if (submitting) {
+      return (
+        <Box flex={{direction: 'row', gap: 8, justifyContent: 'center', alignItems: 'center'}}>
+          <Spinner purpose="body-text" />
+          <div>Evaluating sensor</div>
+        </Box>
+      );
+    } else {
+      return (
+        <Box flex={{direction: 'column', gap: 8}}>
+          <div>Cursor</div>
+          <TextInput
+            value={cursor}
+            onChange={(e) => setCursor(e.target.value)}
+            data-testid={testId('cursor-input')}
+          />
+        </Box>
+      );
+    }
+  }, [
+    sensorExecutionData,
+    error,
+    submitting,
+    currentCursor,
+    cursorState,
+    onPersistCursorValue,
+    name,
+    jobName,
+    repoAddress,
+    cursor,
```

Comment:
Is it worth memoizing here, given the number of deps?

Reply:
I mentioned this on the other PR but I'd agree this isn't worth it / is probably invalidating every render.

It seems like there are some useMemo's in here that are primarily functions for control flow (eg: early returns) and not because they're actually something that needs to be memoized. That might be fine but I think we could also break this component up into a few smaller components (eg: SensorDryRunDialogContent) to achieve the same thing.

Reply:
Memoizing still has some value in case the parent component is needlessly re-rendering, but agreed in this case it doesn't look as good.